src/trace/trace.c

```diff
@@ -1,19 +1,31 @@
+/*
+ * Copyright (c) Valkey Contributors
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
 /* ==========================================================================
  * trace.c - support generic tracing layers.
  * --------------------------------------------------------------------------
- * Copyright (C) 2025  zhenwei pi <pizhenwei@bytedance.com>
+ * Copyright (C) 2025  zhenwei pi <zhenwei.pi@linux.dev>
  * Copyright (C) 2025  zhiqiang li <lizhiqiang.sf@bytedance.com>
  *
  * This work is licensed under BSD 3-Clause, License 1 of the COPYING file in
  * the top-level directory.
  * ==========================================================================
  */
-/*
- * Copyright (c) Valkey Contributors
- * All rights reserved.
- * SPDX-License-Identifier: BSD-3-Clause
- */
 
 #include "trace.h"
 
-int trace_enabled = 0;
+#ifdef USE_LTTNG
+pid_t do_fork(void) {
+    sigset_t sigset;
```

Comment:
in addition, you may want to save errno after fork() returns, and restore it after calling the after_fork functions, just like the lttng-ust fork implementation does.